packages/eslint-plugin-office-addins/src/rules/load-sync-read.ts

```diff
@@ -0,0 +1,81 @@
+import { TSESTree } from "@typescript-eslint/experimental-utils";
+import { Variable } from "@typescript-eslint/experimental-utils/dist/ts-eslint-scope";
+import { findReferences, OfficeApiReference } from "../utils";
+
+export = {
+  name: "load-sync-read",
+  meta: {
+    type: <"problem" | "suggestion" | "layout">"suggestion",
+    messages: {
+      loadSyncRead:
+        "Call load on '{{name}}' for '{{loadValue}}' followed by context.sync() before reading the object or its properties",
+    },
+    docs: {
+      description:
+        "Always call load on an object followed by a sync before reading it or one of its properties.",
+      category: <
+        "Best Practices" | "Stylistic Issues" | "Variables" | "Possible Errors"
+      >"Best Practices",
+      recommended: <false | "error" | "warn">false,
+      url: "https://docs.microsoft.com/en-us/office/dev/add-ins/develop/application-specific-api-model?view=powerpoint-js-1.1#load",
+    },
+    schema: [],
+  },
+  create: function (context: any) {
```

Comment:
This looks like it trying to track the sync for the object and some loads, but I don't think it needs to be that complex and I don't see how it's tracking the loaded properties.  You just want to track load calls without a sync before reading the property . . . you don't need to track the root object.

Reply:
I do need to check if the object is a proxy (get*) object, that's why I'm tracking this.
I'm not really tracking the property that was loaded. Specific properties should be found by another rule:
`load-object-before-read`

Reply:
Specific property read after load is found by another rule . . . but not the sync after property load but before the read.  Saying that now makes me wonder if the two should be combined (read after load become read after sync after load).

Reply:
I'm also wondering if more rules should be combined.  Keeping them separate allows for more targeted rules to turn on and off and combing them would make for a complex rule (especially if we added a config to allow adjusting exactly what is checked), but we seem to be doing repeated work here and it can get confusing which rule is supposed to track what.  I'm not sure what the answer is for combine or not . . . but we should be targeted while keeping seperate.

Reply:
I guess this rule is still different than the others and can detect the unique error as following:
get -> sync -> load -> read
Which is a unique error that only appear to this rule.